src/main/scala/rocket/CSR.scala

```diff
@@ -688,6 +705,13 @@ class CSRFile(
 
     if (usingUser) {
       read_mapping += CSRs.mcounteren -> read_mcounteren
+      read_mapping += CSRs.menvcfg -> reg_menvcfg.asUInt
+      read_mapping += CSRs.senvcfg -> reg_senvcfg.asUInt
+      read_mapping += CSRs.henvcfg -> reg_henvcfg.asUInt
```

Comment:
I assume henvcfg should be added to the read_mapping only when Hypervisor extension is enabled?

>  Additional CSRs are provided to HS-mode, but
not to VS-mode, to manage two-stage address translation and to control the behavior of a VSmode guest: hstatus, hedeleg, hideleg, hvip, hip, hie, hgeip, hgeie, henvcfg, henvcfgh,
hcounteren, htimedelta, htimedeltah, htval, htinst, and hgatp.

After checking the spec, this sentense only exists for menvcfg

> If U-mode is not supported, then registers menvcfg and menvcfgh do not exist.

Simiarly, senvcfg should be defined in the `usingSupervisor` block as it is possible to have `usingUser` but do not have `usingSupervisor`

Reply:
Thanks for the rigorous review, this made me realize my initial implementation bound these registers to haveBasicCounters